### DISCUSSION TITLE
Maintaining Table State

### DISCUSSION THREAD

**Discussion by hanan-commits-work** [iTwin/iTwinUI-react#574] [Q&A] [answered]
Greetings.
I have a table with selectable rows (checkboxes) that dropdowns on Button Click. When I select on a row, it checks it and loads the Reality Model the row is associated with. 
The bug in my code is, whenever the table is closed, it does not retain its previous state. Meaning the checked rows are no longer checked. I want the table to retain its state, even if the table is closed as it will show that this reality model is currently active.

Here are the screenshots..
this one is when i select the rows.
![image](https://user-images.githubusercontent.com/89520756/156569800-f2f1b60a-06da-4f00-bc10-c76ff72a5b1e.png)
Now, when the button is closed. and if I reopen it... the rows get unchecked again
![image](https://user-images.githubusercontent.com/89520756/156569969-a623becd-5b5a-439d-86d0-80de08601c8b.png)

I have been following this sample https://itwin.github.io/iTwinUI-react/?path=/story/core-table--controlled-state .

**Comment by bentleyvk** [comment, marked as answer]
Hi, `onSelect` second parameter is `tableState`. You can save it and then pass it to `initialState` prop.

**Reply by hanan-commits-work** [reply]
Hello. I looked into it, and i have trouble understanding how to accomplish it. Can you please guide me?

**Reply by bentleyvk** [reply]
Sure, let me create codesandbox with the example.

**Reply by hanan-commits-work** [reply]
Thank You.

**Reply by bentleyvk** [reply]
Here you go: https://codesandbox.io/s/save-table-state-9k5x57?file=/src/App.tsx.
Try selecting some rows and then hiding unhiding the Table, it will show selected rows after unhiding.

**Reply by hanan-commits-work** [reply]
Thank you. I appreciate it.

**Reply by bentleyvk** [reply]
Why are you using `Table` here to show "Reality Models"?

**Reply by hanan-commits-work** [reply]
I am basically experimenting.
User can turn on multiple reality models at a moment. I was using select before but it seemed easier to turn on many models because table has checkboxes...
However if you have any alternatives that would be simpler and functional, I'd appreciate hearing it

**Reply by bentleyvk** [reply]
So you kind of need to have multiple values selected in Select?

**Reply by hanan-commits-work** [reply]
Yes.
I need to activate selected reality models and detach unselected...
